src/input/reader_to_storage.cc

```diff
@@ -428,6 +437,57 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Array *arra
 
 
 
+StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Tuple *tuple)
+{
+	int arr_size;
+	if ( (arr_size = p.get_array_size()) != -1 ) {
+
+		StorageArray *storage_array = new StorageArray(tuple->size());
+        // check individual keys
+		for ( Type::Record::KeyIter it= tuple->begin(); it != tuple->end(); ++it) {
+        	if ( p.down(it->key_index) ) {
+                // key on input => check & use it
+                StorageBase *storage = make_storage(p, it->type_.get());
+                if ( (typeid(*storage) == typeid(StorageNull)) && it->default_.has_value_at_declaration() ) {
+                	storage_array->new_item(it->key_index, make_storage_from_default( it->default_.value(), it->type_ ) );
```

Comment:
Just set storage pointer. First deallocate its content. Then use just one common call to storage_array->new_item
Check for deallocation also in other checks for the StorageNull.